lib/course/middleware/logger_middleware.rb

```diff
@@ -0,0 +1,20 @@
+require "logger"
+
+module Course
+  module Middleware
+    class LoggerMiddleware
+      attr_reader :app, :logger
+
+      def initialize(app)
+        @app = app
+        @logger = Logger.new($stdout)
+      end
+
+      def call(env)
+        logger.info("#{env['REQUEST_METHOD']} #{env['REQUEST_URI']}")
```

Comment:
So this solution should be improved:
1. How about giving an ability to pass any logger you want. You've hardcoded Logger, but I'd love to use Semantic logger, you give me a text, but I need a json, you return it into stdout, but I want to log in file. So having an ability to set logger, or fallback to the default one is important.
2. Another important thing to understand, that this logger will be needed in the rest of application. I need an ability to call logger from other middlewares and app, if I know that it was injected. Just think a bit, how it can be done.

Reply:
I have some questions, at first i can pass params in constructor middleware, i did not find info about this. i wanted use dry-container to inject logger, but gems pas params through in constructor. Secondly may be i can use just global app logger like
```ruby
def call(env)
  MyApp.logger.info
  ...
end
```